src/opera/executor/ansible.py

```diff
@@ -62,6 +63,10 @@ def run(host, primary, dependencies, artifacts, variables, verbose, workdir):  #
             "-e", "@" + vars_file,
             playbook
         ]
+
+        if validate:
+            cmd.append("--check")
```

Comment:
What worries me here is that Ansible can make changes even though it's in check mode, if tasks either

* disable check mode explicitly
* use modules that support check mode, but have it implemented incorrectly

This also isn't a 100% way of checking validity, since tasks that depend on the output of prior tasks are skipped.

Could we get away with using only `--syntax-check`? This would remove the need for the workers argument everywhere, at least.

Reply:
Aha yes Ansible says:

> When ansible-playbook is executed with --check it will not make any changes on remote systems. Instead, any module instrumented to support ‘check mode’ (which contains most of the primary core modules, but it is not required that all modules do this) will report what changes they would have made rather than making them. Other modules that do not support check mode will also take no action, but just will not report what changes they might have made. Check mode is just a simulation, and if you have steps that use conditionals that depend on the results of prior commands, it may be less useful for you. However it is great for one-node-at-time basic configuration management use cases.

But as you mentioned users can disable check mode in tasks, which would mean that opera would still execute them. So, at the end I used only `--syntax-check` without the ability to set workers when validating.